core/primitives/src/runtime/migration_data.rs

```diff
@@ -11,10 +13,21 @@ pub struct MigrationData {
     pub storage_usage_delta: Vec<(AccountId, u64)>,
     #[cfg(feature = "protocol_feature_fix_storage_usage")]
     pub storage_usage_fix_gas: Gas,
+    #[cfg(feature = "protocol_feature_restore_receipts_after_fix")]
+    pub restored_receipts: ReceiptResult,
 }
 
 impl Debug for MigrationData {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
         f.debug_struct("MigrationData").finish()
     }
 }
+
+#[derive(Debug, Default)]
+pub struct MigrationFlags {
+    // True iff the current block is the first one in the chain with the current version
+    pub is_first_block_of_version: bool,
+    // True iff the current block containing chunk for some specific shard is the first one in the
+    // chain with the current version
```

Comment:
This seems to be only true if first epoch of the current version have block with chunk (and also currently only for shard 0)